doc/user_guide/user_guide.md

```diff
@@ -0,0 +1,204 @@
+# BucketFS Java User Guide
+
+Exasol's [BucketFS](https://docs.exasol.com/administration/on-premise/bucketfs/bucketfs.htm) is a distributed file system that automatically synchronizes files uploaded on one cluster node to all other nodes. It is intended for providing shared configuration, scripts and libraries used in [User Defined Functions](https://docs.exasol.com/database_concepts/udf_scripts.htm) (UDF) mainly.
+
+This project provides an library that abstracts access to Exasol's [BucketFS](https://docs.exasol.com/administration/on-premise/bucketfs/bucketfs.htm). That allows using BucketFS features programmatically without having to deal with the underlying protocol.
+
+## Notation Used in This Document
+
+Syntax definitions in this document are written in [Augmented Backus-Naur Form (ABNF)](https://tools.ietf.org/html/rfc5234).
+
+## Getting BucketFS Java Into Your Project
+
+BucketFS Java is built using [Apache Maven](https://maven.apache.org/), so integrating the release packages into your project is easy with Maven.
+
+Please check out ["Introduction to the Dependency Mechanism"](http://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html), if you want to learn about how maven handles dependencies and dependency scopes.
+
+We assume here that you are familiar with the basics.
+
+### Exasol Test Containers as Maven Dependency
+
+Just add the following dependency to add the Exasol test containers to your project.
+
+```xml
+<dependency>
+    <groupId>com.exasol</groupId>
+    <artifactId>bucketfs-java</artifactId>
+    <version><!-- add latest version here --></version>
+</dependency>
+```
+
+As always, check for the latest version of the dependencies.
+
+## Services, Buckets, Objects and Paths
+
+A "bucket" in BucketFS terms is a storage container that can contain multiple objects. Buckets are hosted by BucketFS services. They allowing access to a bucket via a network protocol.
+
+To work with BucketFS you need at least one service that hosts one bucket.
+
+An object is roughly comparable to a file. In fact on cluster nodes BucketFS represents objects as files. Note though that this is an internal detail and could be subject to change at any time.
+An object has content and a path.
+
+The full path consists of:
+
+* name of IP or hostname of the machine on which the service runs
+* name of the service hosting the bucket
+* name of the bucket itself
+* relative path to the object inside the bucket
+
+## Object Synchronization
+
+As mentioned before, BucketFS is a distributed file system. Due to that fact, it takes a while until objects uploaded to a bucket on one cluster node are distributed to all other cluster nodes.
+
+Note that to safely use the objects, you need to wait until that synchronization is done.
```

Comment:
Maybe you could tell that this lib helps with that

Reply:
Added.